modules/account-lib/src/coin/hbar/tokenAssociateBuilder.ts

```diff
@@ -0,0 +1,120 @@
+import { BaseCoin as CoinConfig } from '@bitgo/statics';
+import * as proto from '@hashgraph/proto';
+import { BuildTransactionError, TransactionType } from '@bitgo/sdk-core';
+import { Transaction } from './transaction';
+import {
+  buildHederaAccountID,
+  buildHederaTokenID,
+  getHederaTokenIdFromName,
+  isValidAddress,
+  isValidHederaTokenID,
+  stringifyAccountId,
+  stringifyTokenId,
+} from './utils';
+import { TransactionBuilder } from './transactionBuilder';
+
+export class TokenAssociateBuilder extends TransactionBuilder {
+  private readonly _txBodyData: proto.TokenAssociateTransactionBody;
+  private _account: string;
+  private _tokens: string[] = [];
+
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+    this._txBodyData = new proto.TokenAssociateTransactionBody();
+    this._txBody.tokenAssociate = this._txBodyData;
+  }
+
+  /** @inheritdoc */
+  initBuilder(tx: Transaction): void {
+    super.initBuilder(tx);
+    const tokenAssociateAccount = tx.txBody.tokenAssociate;
+    if (tokenAssociateAccount && tokenAssociateAccount.tokens) {
+      this.initTokenAssociation(tokenAssociateAccount as proto.TokenAssociateTransactionBody);
+    }
+  }
+
+  private initTokenAssociation(tokenAssociateAccount: proto.ITokenAssociateTransactionBody): void {
+    tokenAssociateAccount.tokens!.forEach((tokenId: proto.ITokenID) => {
+      const token = stringifyTokenId(tokenId);
+      this.validateToken(token);
+      this._tokens.push(token);
+    });
+
+    if (tokenAssociateAccount.account) {
+      const accountId = stringifyAccountId(tokenAssociateAccount.account);
+      this.account(accountId);
+    }
+  }
+
+  /** @inheritdoc */
+  protected async buildImplementation(): Promise<Transaction> {
+    this._txBodyData.tokens = this.buildTokenData();
+    this._txBodyData.account = this.buildAccountData();
+    this.transaction.setTransactionType(TransactionType.AssociatedTokenAccountInitialization);
```

Comment:
Is this still required since you already set it in `fromRawTransaction` similar for other builder